src/matrix/wishart.jl

```diff
@@ -105,18 +125,21 @@ function meanlogdet(d::Wishart)
 end
 
 function entropy(d::Wishart)
+    d.singular && throw(ArgumentError("entropy not defined for singular Wishart."))
     p = dim(d)
     df = d.df
     -d.logc0 - 0.5 * (df - p - 1) * meanlogdet(d) + 0.5 * df * p
 end
 
 #  Gupta/Nagar (1999) Theorem 3.3.15.i
 function cov(d::Wishart, i::Integer, j::Integer, k::Integer, l::Integer)
+    d.singular && throw(ArgumentError("cov not defined for singular Wishart."))
```

Comment:
Can you check numerically if this holds for singular too, like for the mean?

Reply:
Yeah, the derivation is the same, and it checks out in simulations.